code/datums/components/barber.dm

```diff
@@ -0,0 +1,605 @@
+		thing.dir = SOUTH
+		return
+	thing.force_use_as_tool = !thing.force_use_as_tool
+	if (thing.force_use_as_tool)
+		thing.dir = WEST
+	else
+		thing.dir = SOUTH
+
+	var/list/cool_grip_adj = list("a sick", "a wicked", "a deadly", "a menacing", "an edgy", "a tacticool", "a sweaty", "an awkward")
+	var/list/cool_grip1 = list("combat", "fightlord", "guerilla", "hidden", "space", "syndie", "double-reverse", "\"triple-dog-dare-ya\"", "stain-buster's")
+	var/list/cool_grip2a = list("blade", "cyber", "street", "assistant", "comedy", "butcher", "edge", "beast", "heck", "crud", "ass")
+	var/list/cool_grip2b = list("master", "slayer", "fighter", "militia", "space", "syndie", "lord", "blaster", "beef", "tyrannosaurus")
+	var/list/wheredWeSeeIt = list("saw the clown do", "saw the captain do", "saw the head of security do",\
+														"saw someone in a red spacesuit do", "saw a floating saw do", "saw on TV",\
+														"saw one of the diner dudes do", "saw just about every assistant do")
+	var/cool_grip3 = "[pick(wheredWeSeeIt)] [pick("once", "once or twice")]"
+
+	if(thing.force_use_as_tool)
+		user.visible_message("[user] assumes a less hostile grip on the [thing].",\
+													"You change your grip on the [thing], so as to use it more as a tool than a weapon.")
+	else
+		user.visible_message("[user] wields the [thing] a with [pick(cool_grip_adj)] [pick(cool_grip1)] [pick(cool_grip2a)][pick(cool_grip2b)] [pick("style", "grip")] that they probably [pick(cool_grip3)]!",\
+													"You wield the [thing] with [pick(cool_grip_adj)] [pick(cool_grip1)] [pick(cool_grip2a)][pick(cool_grip2b)] [pick("style", "grip")] that you [pick(cool_grip3)]! It makes it just about impossible to use as a tool!")
+
+/datum/component/toggle_tool_use/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_ITEM_ATTACK_SELF)
+	UnregisterSignal(parent, COMSIG_ITEM_DROPPED)
+	. = ..()
+
+/datum/component/barber
+/datum/component/barber/Initialize()
+	if(!istype(parent, /obj/item))
+		return COMPONENT_INCOMPATIBLE
+
+/datum/component/barber/haircut
+/datum/component/barber/haircut/Initialize()
+	RegisterSignal(parent, list(COMSIG_ITEM_ATTACK_PRE), .proc/do_haircut)
+
+/datum/component/barber/shave
+/datum/component/barber/shave/Initialize()
+	RegisterSignal(parent, list(COMSIG_ITEM_ATTACK_PRE), .proc/do_shave)
+
+/datum/component/barber/proc/do_haircut(var/obj/item/thing, mob/living/carbon/human/M as mob, mob/living/carbon/human/user as mob)
+	if(!M || !user || (user.a_intent != INTENT_HELP && !thing.force_use_as_tool))
+		return 0 // Who's cutting whose hair, now?
+
+	var/mob/living/carbon/human/H = M
+	if(ishuman(M) && ((H.head && H.head.c_flags & COVERSEYES) || (H.wear_mask && H.wear_mask.c_flags & COVERSEYES) || (H.glasses && H.glasses.c_flags & COVERSEYES)))
+		// you can't stab someone in the eyes wearing a mask!
+		boutput(user, "<span class='notice'>You're going to need to remove that mask/helmet/glasses first.</span>")
+		. = BARBERY_RESOLVABLE
```

Comment:
When it comes to this and similar lines below - is BARBERY_RESOLVABLE actually checked anywhere? It seems like this proc should only return `ATTACK_PRE_DONT_ATTACK` or `0` probably?

Reply:
BARBERY_RESOLVABLE means that barbery failed, but gave a useful message about why. Should be checked lower down, first if (.) is not 0, then if (.) != BARBERY_RESOLVABLE, throw a generic message and return ATTACK_PRE_DONT_ATTACK.

Reply:
But you also do `return BARBERY_SUCCESSFUL` later.

Reply:
I'd just use a new var for these instead of `.`. When I see `. = stuff` I assume you are planning to return `stuff`.

Reply:
Aight, replaced them with a var.

Also it turned out that I didnt actually need most of those defines. Aside from the initial pre-checks, they werent even used! All that got cut and replaced with a var.